warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -961,142 +859,43 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
         TraceStopwatch stopwatch = null;
         
         if (!disableExpandIndexFunction) {
-            stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand function index queries");
-            
             // expand the index queries for the functions
-            queryTree = FunctionIndexQueryExpansionVisitor.expandFunctions(config, metadataHelper, dateIndexHelper, queryTree);
-            if (log.isDebugEnabled()) {
-                logQuery(queryTree, "Query after function index queries were expanded:");
-            }
-            
-            stopwatch.stop();
+            queryTree = timedExpandIndexQueriesForFunctions(timers, queryTree, config, metadataHelper);
         }
         
         // apply the node transform rules
         // running it here before any unfielded expansions to enable potentially pushing down terms before index lookups
-        queryTree = applyNodeTransformRules(queryTree, getTransformRules(), config, metadataHelper, "Pre unfielded expansions");
+        queryTree = timedApplyNodeTransformRules(timers, "Apply Node Transform Rules - Pre Unfielded Expansions", queryTree, config, metadataHelper,
+                        getTransformRules());
         
         // Find unfielded terms, and fully qualify them with an OR of all fields
         // found in the index
         // If the max term expansion is reached, then the original query tree is
         // returned.
         // If the max regex expansion is reached for a term, then it will be
         // left as a regex
-        Set<String> expansionFields = null;
+        Set<String> expansionFields = loadExpansionFields(config);
```

Comment:
this should only be executed if !disabledAnyFieldLookup

Reply:
done